x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/utils/ExceptionCollectionHandling.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ *
+ * This file was contributed to by generative AI
+ */
+
+package org.elasticsearch.xpack.ml.utils;
+
+import org.elasticsearch.ElasticsearchStatusException;
+import org.elasticsearch.common.util.concurrent.AtomicArray;
+import org.elasticsearch.rest.RestStatus;
+
+import java.util.List;
+
+public class ExceptionCollectionHandling {
+
+    public static ElasticsearchStatusException exceptionArrayToStatusException(AtomicArray<Exception> failures, String message) {
+
+        List<Exception> caughtExceptions = failures.asList();
+        if (caughtExceptions.isEmpty()) {
+            throw new ElasticsearchStatusException("No exceptions caught", RestStatus.INTERNAL_SERVER_ERROR);
+        } else {
+
+            boolean allElasticsearchStatusException = true;
```

Comment:
This logic can work on `ElasticsearchException` rather than `ElasticsearchStatusException`, because `ElasticsearchException` has the `status` method.